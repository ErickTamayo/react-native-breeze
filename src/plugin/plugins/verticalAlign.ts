import { PluginFunction } from "./types";

export type PluginGroups = { alignment: "auto" | "top" | "bottom" | "center" };

export const pattern = /^align-(?<alignment>auto|top|bottom|center)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { alignment } = groups;
  return { textAlignVertical: alignment };
};
