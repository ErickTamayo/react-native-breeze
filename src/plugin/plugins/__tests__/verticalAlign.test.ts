import { pattern, plugin } from "../verticalAlign";
import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";

describe("verticalAlign", () => {
  // prettier-ignore
  const shouldMatch = generateInput("align", ["auto", "top", "bottom", "center"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["align-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
