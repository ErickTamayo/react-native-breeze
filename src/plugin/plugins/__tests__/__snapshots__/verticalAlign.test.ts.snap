// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verticalAlign Should parse (align-auto) correctly 1`] = `
Object {
  "textAlignVertical": "auto",
}
`;

exports[`verticalAlign Should parse (align-bottom) correctly 1`] = `
Object {
  "textAlignVertical": "bottom",
}
`;

exports[`verticalAlign Should parse (align-center) correctly 1`] = `
Object {
  "textAlignVertical": "center",
}
`;

exports[`verticalAlign Should parse (align-top) correctly 1`] = `
Object {
  "textAlignVertical": "top",
}
`;
